src/Microsoft.OData.Client/DeepInsertSaveResult.cs

```diff
@@ -118,6 +120,65 @@ internal void DeepInsertRequest<T>(T resource)
             }
         }
 
+        /// <summary>
+        /// Asynchronous deep insert request.
+        /// </summary>
+        /// <typeparam name="T">The type of the top-level object to be deep inserted.</typeparam>
+        /// <param name="resource">The top-level object of the type to be deep inserted.</param>
+        internal void BeginDeepInsertRequest<T>(T resource)
+        {
+            PerRequest pereq = null;
+
+            if (resource == null)
+            {
+                throw Error.ArgumentNull(nameof(resource));
+            }
+
+            BuildDescriptorGraph(this.ChangedEntries, true, resource);
+
+            try
+            {
+                ODataRequestMessageWrapper deepInsertRequestMessage = this.GenerateDeepInsertRequest();
+                this.Abortable = deepInsertRequestMessage;
+
+                if (deepInsertRequestMessage != null)

```

Comment:
@KenitoInc Under which conditions can `deepInsertRequestMessage` be null? Do you have a test for such a scenario? This question gains further relevant due to the call to `HandleCompleted` method in the finally block where we pass `pereq` - it'd be null in that scenario

Reply:
Refactored that code